solr/core/src/java/org/apache/solr/cloud/gumi/CreateCollectionRequest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.cloud.gumi;
+
+/**
+ * <p>Request for creating a new collection with a given number of shards and replication factor for various replica types.</p>
+ *
+ * Note there is no need at this stage to allow the request to convey each shard hash range for example, this can be handled
+ * by the Solr side implementation without needing the plugin to worry about it.
+ *
+ */
+public interface CreateCollectionRequest extends Request {
+  String getCollectionName();
+
+  int getShardCount();
+
+  int getNRTReplicationFactor();
```

Comment:
`getNrtReplicaCount()` ? and so on ?

Reply:
Not sure we need the count here. This is a creation request for a new collection, not a representation of the cluster state.
If you look at the `[Shard](https://github.com/apache/lucene-solr/pull/1684/files#diff-9633f5e169fa3095062451599daac213)` class, that's where cluster state is passed to the plugin and there we do have access to the replicas.